test/e2e/framework/util.go

```diff
@@ -3956,6 +3972,20 @@ func isNodeConditionSetAsExpected(node *v1.Node, conditionType v1.NodeConditionT
 		// Ensure that the condition type and the status matches as desired.
 		if cond.Type == conditionType {
 			if (cond.Status == v1.ConditionTrue) == wantTrue {
+				if cond.Type != v1.NodeReady {
```

Comment:
regarding previous line: i believe if cond.Status == false && wantTrue == false, it will check for taints, which it shouldn't be doing